libusb/os/linux_usbfs.c

```diff
@@ -1438,7 +1438,7 @@ static void op_close(struct libusb_device_handle *dev_handle)
 
 	/* fd may have already been removed by POLLERR condition in op_handle_events() */
 	if (!hpriv->fd_removed)
-		usbi_remove_event_source(HANDLE_CTX(dev_handle), hpriv->fd);
+		usbi_remove_event_source(DEVICE_CTX(dev_handle->dev), hpriv->fd);
```

Comment:
Personally, I'm not a fan of this change.  The old formulation is more clear in its intent.

You said an explicit null check did not pacify the tool, but what about a `LIBUSB_NONNULL(1)` attribute on op_close?

Reply:
Hmm, yes, that might work. Maybe just do that and see if anyone complains in the future? Or just drop this :-)

I am not in a position to rerun the checker anymore. I was doing this work for the RHEL libusb package but I am not with Red Hat anymore. As such, the work I can do on this is limited, but I am happy to do a quick rebase to drop the commits that don't seem quite right.

Though, also feel free to just cherry-pick whatever you like, either push to this PR or open a new one.